starfish/test/spots/detector/test_spot_detection.py

```diff
@@ -74,19 +78,30 @@ def test_spot_detection_with_reference_image(
         max_intensity: float,
 ):
     """This testing method uses a reference image to identify spot locations."""
-    reference_image_mp = data_stack.max_proj(Axes.CH, Axes.ROUND)
-    reference_image_mp_numpy = reference_image_mp._squeezed_numpy(Axes.CH, Axes.ROUND)
-
-    intensity_table = detect_spots(data_stack=data_stack,
-                                   spot_finding_method=spot_detector.image_to_spots,
-                                   reference_image=reference_image_mp_numpy,
-                                   measurement_function=np.max,
-                                   radius_is_gyration=radius_is_gyration)
+
+    def call_detect_spots(stack):
+
+        reference_image_mp = stack.max_proj(Axes.CH, Axes.ROUND)
+        reference_image_mp_numpy = reference_image_mp._squeezed_numpy(Axes.CH, Axes.ROUND)
+
+        return detect_spots(
+            data_stack=stack,
+            spot_finding_method=spot_detector.image_to_spots,
+            reference_image=reference_image_mp_numpy,
+            measurement_function=np.max,
+            radius_is_gyration=radius_is_gyration
+        )
+
+    intensity_table = call_detect_spots(data_stack)
```

Comment:
why is this refactor necessary? def call_detect_spots

Reply:
It's being called both with `data_stack` and `EMPTY_IMAGESTACK`. This route introduces a function that defines the parameters only once.